components/weekly_storage/weekly_storage_unittest.cc

```diff
@@ -47,6 +47,24 @@ TEST_F(WeeklyStorageTest, AddsSavings) {
   EXPECT_EQ(state_->GetWeeklySum(), saving * 3);
 }
 
+TEST_F(WeeklyStorageTest, SubDelta) {
+  state_->AddDelta(5000);
+  clock_->Advance(base::Days(1));
+  state_->AddDelta(3000);
+  clock_->Advance(base::Days(1));
+  state_->AddDelta(1000);
+  clock_->Advance(base::Days(1));
+
+  state_->SubDelta(500);
+  EXPECT_EQ(state_->GetWeeklySum(), 8500U);
+  state_->SubDelta(4000);
+  EXPECT_EQ(state_->GetWeeklySum(), 4500U);
+  state_->SubDelta(5000);
+  EXPECT_EQ(state_->GetWeeklySum(), 0U);
```

Comment:
Please add a comment here about how the overall sum can't go negative and further. Also please advance far enough for an earlier `AddDelta` value to expire and show that it still saturates properly.